### PR TITLE
✨ add script to insert demo/test data

### PR DESCRIPTION
**Pull request by eliascodes**
Related to #338 

This is part of moving towards a full TS stack (and retiring JS files/scripts).

This is implementing the beginning of the structure I sketched out in my comment: https://github.com/TwinePlatform/twine-api/issues/338#issuecomment-462345624

* Added brief docs explaining structure of test data
* Re-organise how seed and test data is stored and inserted into the DB
* Upgrade `jest` so that `global_setup` can be converted to TS
* Add simple CLI entry point to insert test/demo data (to be improved in future)

I had much more significant changes in mind, but figured I shouldn't run before I can walk.

### CODE REVIEW

**Comment by eliascodes** [comment]
User testing:
* [ ] Reset local (dev) DB
   1. Run `npm run db:reset` -> Should have no data in DB (except gender, disability, etc)
   2. Run `npm exec bin/insert_data.ts demo` -> All demo data should be inserted
* [ ] Reset remote (staging?) DB (same steps as above, except setting `NODE_ENV`)
* [ ] Run tests

**Comment by astroash** [comment]
**Manual testing feedback:**
- if we are using `exec` in this way it would be nice to get more visual feedback of what is happening eg rows inserted, complete ect. Currently the only feedback is:
```
> twine-api@1.0.0 exec <directory>/twine/twine-api
> ts-node --files "bin/insert_data.ts" "demo"
```
- migrations suggestion outlined in https://github.com/TwinePlatform/twine-api/issues/338#issuecomment-464654553

**Comment by eliascodes** [comment]
> migrations suggestion outlined in [#338 (comment)](https://github.com/TwinePlatform/twine-api/issues/338#issuecomment-464654553)

I like this idea but I think it's a bigger piece of work.

**Comment by astroash** [comment]
> I like this idea but I think it's a bigger piece of work

Are we in any rush to get this merged?

**Comment by eliascodes** [comment]
Well, actually having the seed data on master/staging might be nice. Beyond that, I guess not. But also depends on how you feel about reviewing enormous PRs 🙂

**Comment by astroash** [comment]
> But also depends on how you feel about reviewing enormous PRs 🙂

😅 I can make an exception.

I feel like this PR is v close to having a super sweet db setup. As all the changes are to scripts that we have to run manually I see little benefit of it being merged to staging/master. I am happy to wait for this to be amended to run as migrations.

That said, if you think the migration work is less of a priority than other things on our backlog then we can do it in stages (eg merge this now then get to migrations later). First set of demo data has already been added to production so we will have to fake the migrations table for those entries anyway.

**Comment by astroash** [comment]
@eliascodes is this ready for review?

**Comment by eliascodes** [comment]
I decided to remove the extra work because it's better to do it bit by bit to prevent me from losing my mind 🙃.

It's now at the same state as it was when you originally left your review, (except with some logging added).

File: bin/insert_data.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * DB CLI
+ */
```
Comment (astroash) [inline bin/insert_data.ts]: can you add instructions on how this works

**Comment by astroash** [comment]
@eliascodes please update the PR body and squash-merge 🙂

**Pull request merged**